connector/spanmetricsconnector/connector_test.go

```diff
@@ -780,33 +758,99 @@ func newConnectorImp(mcon consumer.Metrics, defaultNullValue *pcommon.Value, tem
 	}
 }
 
-func TestUpdateExemplars(t *testing.T) {
-	// ----- conditions -------------------------------------------------------
+func TestConnector_AggregateLatencies(t *testing.T) {
 	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig().(*Config)
+
 	traces := buildSampleTrace()
-	traceID := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0).TraceID()
-	spanID := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0).SpanID()
-	key := metricKey("metricKey")
+	span := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0)
+
+	// aggregate over the same metric key
 	c, err := newConnector(zaptest.NewLogger(t), cfg, nil)
-	c.metricsConsumer = new(consumertest.MetricsSink)
-	value := float64(42)
+	require.NoError(t, err)
 
-	// ----- call -------------------------------------------------------------
-	h := c.getOrCreateHistogram(key, pcommon.NewMap())
-	h.observe(value, traceID, spanID)
+	c.aggregateLatencies("key", span.Attributes(), span, 1.5)
+	c.aggregateLatencies("key", span.Attributes(), span, 1.7)
+
+	got, ok := c.histograms["key"]
+	assert.True(t, ok)
```

Comment:
Same with others below, because other assertions would be invalid if `ok == false`.

```suggestion
	require.True(t, ok)
```